frontend/server/blockSerialization.ts

```diff
@@ -0,0 +1,309 @@
+import { spawnAsync } from "./spawnAsync";
```

Comment:
The file extension changed so it says the whole file changed, but I think you only modified callAgent so I'm only going to look at that. Also wondering why we need typescript is it just for type checking the inputs or we need a a library that's TS only?

Reply:
if i'm going to write tests, i'm going to write types.